editor/src/components/canvas/commands/wrap-in-conditional-command.ts

```diff
@@ -0,0 +1,118 @@
+import {
+  emptyComments,
+  jsExpressionValue,
+  jsxConditionalExpression,
+  JSXElementChild,
+  jsxFragment,
+} from '../../../core/shared/element-template'
+import { ElementPath } from '../../../core/shared/project-file-types'
+import {
+  EditorState,
+  EditorStatePatch,
+  forUnderlyingTargetFromEditorState,
+  insertElementAtPath,
+  removeElementAtPath,
+} from '../../editor/store/editor-state'
+import { BaseCommand, CommandFunction, getPatchForComponentChange, WhenToRun } from './commands'
+import * as EP from '../../../core/shared/element-path'
+import { getUtopiaJSXComponentsFromSuccess } from '../../../core/model/project-file-utils'
+import { InsertionSubjectWrapper } from '../../editor/editor-modes'
+import { assertNever } from '../../../core/shared/utils'
+
+type ContainerToWrapIn = InsertionSubjectWrapper
+
+export interface WrapInContainerCommand extends BaseCommand {
+  type: 'WRAP_IN_CONDITIONAL'
+  whenToRun: WhenToRun
+  target: ElementPath
+  wrapperUID: string
+  wrapper: ContainerToWrapIn
+}
+
+export function wrapInContainerCommand(
+  whenToRun: WhenToRun,
+  target: ElementPath,
+  wrapperUID: string,
+  wrapper: ContainerToWrapIn,
+): WrapInContainerCommand {
+  return {
+    type: 'WRAP_IN_CONDITIONAL',
+    whenToRun: whenToRun,
+    target: target,
+    wrapperUID: wrapperUID,
+    wrapper: wrapper,
+  }
+}
+
+export const runWrapInConditionalCommand: CommandFunction<WrapInContainerCommand> = (
+  editor: EditorState,
+  command: WrapInContainerCommand,
+) => {
+  let editorStatePatches: Array<EditorStatePatch> = []
+
+  forUnderlyingTargetFromEditorState(
+    command.target,
+    editor,
+    (success, elementToWrap, _underlyingTarget, underlyingFilePath) => {
+      const components = getUtopiaJSXComponentsFromSuccess(success)
+      const withElementRemoved = removeElementAtPath(command.target, components)
+
+      const wrapper = getInsertionSubjectWrapper(command.wrapper, command.wrapperUID, elementToWrap)
+
+      // Insert the conditional at the initial index
+      const targetParent = EP.parentPath(command.target)
+      const insertionResult = insertElementAtPath(
+        editor.projectContents,
+        underlyingFilePath,
+        targetParent,
+        wrapper,
+        withElementRemoved,
+        null, // FIXME Find the index position of the original element
```

Comment:
Aha yes, this needs fixing, otherwise it will break for insertions that respect the index position (so e.g. inserting into a flex container in between two elements would result in the new element being inserted at the end)

Reply:
> e.g. inserting into a flex container in between two elements

sounds like this should be a test case as well then!

Reply:
I'll add this on a separate PR, since this might be a little involved